test/components/componentTests.ts

```diff
@@ -246,333 +344,338 @@ describe("Component behavior", () => {
-    c.anchor(svg);
-    c._setup();
-    c.render();
-    assert.isFalse(renderFlag, "no render until width/height set to nonzero");
-
-    c._width = 10;
-    c._height = 0;
-    c.render();
-    assert.isTrue(renderFlag, "render still occurs if one of width/height is zero");
+  describe("destroying", () => {
+    it("cannot reanchor if destroyed", () => {
+      c.renderTo(svg);
+      c.destroy();
 
-    c._height = 10;
-    c.render();
-    assert.isTrue(renderFlag, "render occurs if width and height are positive");
+      assert.throws(() => c.renderTo(svg), "reuse");
+      svg.remove();
+    });
 
-    svg.remove();
+    it("performs all of the same operations as detaching", () => {
+      let detachCalled = false;
+      c.detach = () => {
+        detachCalled = true;
+        return c;
+      };
+      c.destroy();
+      assert.isTrue(detachCalled, "detach called in destroy invocation");
+      svg.remove();
+    });
   });
 
-  it("rendering to a new svg detaches the component", () => {
-    let SVG_HEIGHT_1 = 300;
-    let SVG_HEIGHT_2 = 50;
-    let svg1 = TestMethods.generateSVG(300, SVG_HEIGHT_1);
-    let svg2 = TestMethods.generateSVG(300, SVG_HEIGHT_2);
-
-    let plot = new Plottable.Plots.Line();
-    let group = new Plottable.Components.Group;
-    plot.x(0).y(0);
-    group.renderTo(svg1);
-
-    group.append(plot);
-
-    assert.deepEqual(plot.parent(), group, "the plot should be inside the group");
-    assert.strictEqual(plot.height(), SVG_HEIGHT_1, "the plot should occupy the entire space of the first svg");
-
-    plot.renderTo(svg2);
+  describe("rendering on the anchored svg", () => {
+    it("performs all of the same operations as rendering immediately", () => {
+      let renderFlag = false;
+      let c = new Plottable.Component();
+      c.renderImmediately = () => {
+        renderFlag = true;
+        return c;
+      };
+      c.anchor(svg);
+      c.computeLayout();
+      assert.strictEqual(c.render(), c, "returns calling object");
+      assert.isTrue(renderFlag, "renderImmediately called in render invocation");
 
-    assert.strictEqual(plot.parent(), null, "the plot should be outside the group");
-    assert.strictEqual(plot.height(), SVG_HEIGHT_2, "the plot should occupy the entire space of the second svg");
+      c.destroy();
+      svg.remove();
+    });
 
-    svg1.remove();
-    svg2.remove();
-    svg.remove();
+    it("does not render unless allocated space", () => {
+      let renderFlag = false;
+      let c = new Plottable.Component();
+      c.renderImmediately = () => {
+        renderFlag = true;
+        return c;
+      };
+      c.anchor(svg);
+      c.render();
+      assert.isFalse(renderFlag, "no render until width/height set to nonzero");
+
+      let offeredWidth = 10;
+      let offeredHeight = 0;
+      c.computeLayout({x: 0, y: 0}, offeredWidth, offeredHeight);
+      c.render();
+      assert.isTrue(renderFlag, "render still occurs if one of width/height is zero");
```

Comment:
Note that `render` has been used more semantically here more than code-wise, since I did not mock out the `render` function.